Toshi/Source/Toshi/File/TTRB.cpp

```diff
@@ -67,32 +67,22 @@ namespace Toshi
 					TTODO("HEAD section");
 
 					int numsections = (sectionSize - 4) / 0xC;
-
-					m_pHeader = static_cast<Header*>(tmalloc(numsections * 0x10 + 8));
-
+					m_pHeader = static_cast<Header*>(tmalloc(sizeof(Header) + sizeof(SecInfo) * numsections));
 					m_pHeader->m_ui32Version = 0;
 
 					ttsf.ReadBytes(&m_pHeader->m_i32SectionCount, sizeof(m_pHeader->m_i32SectionCount));
 
-					TASSERT(m_pHeader->m_ui32Version == TMAKEVERSION(0, 0), "");
-					TASSERT(m_pHeader->m_i32SectionCount == numsections, "");
-
-					char* cursor = (char*)(m_pHeader + 1);
+					// TASSERT(m_pHeader->m_ui32Version == TMAKEVERSION(0, 0), "HEAD section cannot have a version");
```

Comment:
is that TASSERT neccessary since m_ui32Version is set to 0 at line 71?

Reply:
No, i just did it cause the original did it. But it is really useless and it's also worse performance